inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/mq/DeletePulsarResourceTaskListener.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.inlong.manager.service.mq.util.PulsarUtils;
+import org.apache.inlong.manager.workflow.WorkflowContext;
+import org.apache.inlong.manager.workflow.event.ListenerResult;
+import org.apache.inlong.manager.workflow.event.task.QueueOperateListener;
+import org.apache.inlong.manager.workflow.event.task.TaskEvent;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Delete Pulsar tenant, namespace and topic after group is operated to delete
+ */
+@Slf4j
+@Component
+public class DeletePulsarResourceTaskListener implements QueueOperateListener {
+
+    @Autowired
+    private InlongGroupService groupService;
+    @Autowired
+    private InlongClusterService clusterService;
+    @Autowired
+    private PulsarOperator pulsarOperator;
+
+    @Override
+    public TaskEvent event() {
+        return TaskEvent.COMPLETE;
+    }
+
+    @Override
+    public ListenerResult listen(WorkflowContext context) throws Exception {
+        GroupResourceProcessForm form = (GroupResourceProcessForm) context.getProcessForm();
+        String groupId = form.getInlongGroupId();
+        log.info("begin to delete pulsar resource for groupId={}", groupId);
+        InlongGroupInfo groupInfo = groupService.get(groupId);
+        if (groupInfo == null) {
+            throw new WorkflowListenerException("inlong group or pulsar cluster not found for groupId=" + groupId);
+        }
+        try {
+            InlongPulsarInfo pulsarInfo = (InlongPulsarInfo) groupInfo;
+            InlongClusterInfo clusterInfo = clusterService.getOne(groupInfo.getInlongClusterTag(), null,
+                    ClusterType.CLS_PULSAR);
+            deletePulsarProcess(pulsarInfo, (PulsarClusterInfo) clusterInfo);
+        } catch (Exception e) {
+            log.error("delete pulsar resource error for groupId={}", groupId, e);
+            throw new WorkflowListenerException("delete pulsar resource error for groupId=" + groupId);
+        }
+
+        log.info("success to delete pulsar resource for groupId={}", groupId);
+        return ListenerResult.success();
+    }
+
+    /**
+     * Delete Pulsar namespace forcefully
+     */
+    private void deletePulsarProcess(InlongPulsarInfo pulsarInfo, PulsarClusterInfo pulsarCluster) throws Exception {
+        String groupId = pulsarInfo.getInlongGroupId();
+        log.info("begin to delete pulsar resource for groupId={} in cluster={}", groupId, pulsarCluster);
+
+        String namespace = pulsarInfo.getMqResource();
+        Preconditions.checkNotNull(namespace, "pulsar namespace cannot be empty for groupId=" + groupId);
+        try (PulsarAdmin pulsarAdmin = PulsarUtils.getPulsarAdmin(pulsarCluster)) {
+            String tenant = pulsarCluster.getTenant();
+            // create pulsar namespace
+            pulsarOperator.forceDeleteNamespace(pulsarAdmin, tenant, namespace);
```

Comment:
The namespace may be used by another user, delete namespace is a dangerous operation.

Reply:
ok